### PR TITLE
add image-processor to project create-edit

### PR DESCRIPTION
**Pull request by nothingismagick**
- increase amount of images to 6 (fits better)
- put gallery of images above uploader
- hide image-processor when images = 6
- use jpeg/80% for storing images

### Verify that
Git checklist
- [x] the branch name follows the git flow naming convention + label + UT reference if applicable (feature/i18n/UT-283)
- [x] I have rebased my branch to develop before submitting my PR

Front end checklist
- [x] the design is responsive
- [x] the code doesn't break Server Side Rendering

API checklist
- [X] the code doesn't break the api endpoints contract (applicable when we're in production and the api is officially open and used by 3rd party app)

Tests and comments checklist
- [x] JS Standard code style has been verified with ESLint
- [x] functions are prefixed with JSDoc information or the doc has been updated

### My PR doesn't concern a Jira task
This PR merely adds the image-processor component to the project create/edit page.
---

Please provide a test plan so that the reviewers can see how you tested your work and/or the different acceptance criteria described in the task.

### **Test plan**

_Feature / acceptance criteria tested_

Add Project Avatar
* Step 1 add an image
* Step 2 zoom and move to fit
* Step 3 update the image
* Step 4 update the project
The avatar should be saved with the project

Add Project Images
* Step 1 create a new project
* Step 2 add an image (and all other required fields)
* Step 3 save
- the image should still be there
* Step 4 add 5 more images and save
- the images should still be there
* Step 5 remove an image
- the image-processor component should appear

### CODE REVIEW

File: packages/client/src/pages/projects/create-edit/create-edit.vue [hunk 1]
```diff
@@ -3,20 +3,25 @@ import { mapActions, mapGetters } from 'vuex'
...
   data () {
     return {
-      media5: {
-        ref: 'media5',
+      media: {
+        ref: 'media',
         imageValid: false,
-        width: 2000,
-        height: 1130,
+        width: 512,
```
Comment (gregory-latinier) [inline packages/client/src/pages/projects/create-edit/create-edit.vue]: The size here is now wrong.
The latest mockup for the view page uses images with a width of 770px and a height of 420px.
If you use larger images like twice or three times the size, check / modify the project view page. But you have to keep the ratio.

File: packages/api/modules/projects/validate.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ const createProject = {
     website: Joi.string().optional().trim().max(1000).uri(),
     docs: Joi.string().optional().trim().max(1000).uri(),
     license: Joi.string().trim().required().allow(licenses),
-    medias: Joi.array().min(1).max(5).required(),
+    medias: Joi.array().min(1).max(6).required(),
```
Comment (gregory-latinier) [inline packages/api/modules/projects/validate.js]: I don't mind having 6 images but why did you change it?

File: packages/client/src/pages/projects/create-edit/create-edit.vue [hunk 3]
```diff
@@ -137,37 +142,33 @@ export default {
...
-              this.$refs.uploader.reset()
-              this.uploading = false
-            }).catch(() => {
-              reject(file)
-              this.uploading = false
+      if (this.project.medias.filter(m => m.type === 'image').length >= 6) {
+        this.uploading = false
+      } else {
+        this.$refs[ref].upload('image/jpeg', 0.8).then(async (res) => {
+          console.log('upload', res)
```
Comment (gregory-latinier) [inline packages/client/src/pages/projects/create-edit/create-edit.vue]: 💩

File: packages/client/src/pages/projects/create-edit/create-edit.vue [hunk 4]
```diff
@@ -357,14 +358,30 @@ div
...
+          .float-right
+            q-btn.q-pa-sm.q-ma-sm(color="neutral", text-color="black", v-if="media.imageValid", :label="$t('projects.createEdit.images.clear')", @click="$refs.media.clear()")
+            q-btn.q-pa-sm.q-ma-sm(color="primary", v-if="media.imageValid", :label="$t('projects.createEdit.images.upload')", @click="updateImages('media')")
 
-      q-field(v-if="project.medias.filter(m => m.type === 'image').length > 0")
-        gallery(:images="project.medias.filter(m => m.type === 'image').map((i) => i.src)" :index="galleryIdx" @close="galleryIdx = null")
-        .project-image(v-for="(image, imageIndex) in project.medias.filter(m => m.type === 'image')", :key="imageIndex")
-          q-btn(round, dense, icon="mdi-minus", color="red", size="xs" @click="() => removeMedia(image.src)")
-          div(@click="galleryIdx = imageIndex", :style="{ backgroundImage: 'url(' + image.src + ')' }")
+      // q-uploader(url="", ref="uploader", :upload-factory="uploadFile", @add="(files) => uploadFile(files)", @fail="uploadFails", hide-upload-button,
```
Comment (gregory-latinier) [inline packages/client/src/pages/projects/create-edit/create-edit.vue]: 💩

File: packages/i18n/locales_master/en.json [hunk 5]
```diff
@@ -121,7 +121,9 @@
       },
       "images": {
         "label": "Images",
-        "help": "This is the first thing that people will see when they come across your project. 2000x1130 ideal for hi-res. Select a maximum of 5 images."
+        "help": "This is the first thing that people will see when they come across your project. 2048x1024 ideal for hi-res. Select a maximum of 6 images.",
```
Comment (gregory-latinier) [inline packages/i18n/locales_master/en.json]: Can you change the size in the message too?

File: packages/client/src/pages/projects/create-edit/create-edit.vue [hunk 6]
```diff
@@ -137,37 +142,33 @@ export default {
...
-              resolve(file)
-              this.$refs.uploader.reset()
-              this.uploading = false
-            }).catch(() => {
-              reject(file)
-              this.uploading = false
+      if (this.project.medias.filter(m => m.type === 'image').length >= 6) {
+        this.uploading = false
+      } else {
+        this.$refs[ref].upload('image/jpeg', 0.8).then(async (res) => {
```
Comment (gregory-latinier) [inline packages/client/src/pages/projects/create-edit/create-edit.vue]: Your upload instruction is not the same than the one in the profile page.
Can you update the profile page?
The profile page should use jpeg too instead of png it's lighter.

File: packages/client/src/pages/projects/create-edit/create-edit.vue (hunk 3 above)
Comment (nothingismagick) [reply packages/client/src/pages/projects/create-edit/create-edit.vue]: yuck - too late at night for that commit jeez

File: packages/api/modules/projects/validate.js (hunk 2 above)
Comment (nothingismagick) [reply packages/api/modules/projects/validate.js]: I changed it because in the view only 5 images left an ugly blank space:
```
___ ___ ___
| | | | | |
___ ___ ___
___ ___
| | | |
___ ___

```

Comment (nothingismagick) [reply packages/api/modules/projects/validate.js]: 6 gives us better rows for different sizes depending on the scaling and column width
2+2+2, 3+3, 4+2

**Comment by icaroharry** [comment]
I'm having this error when trying to upload an image:
image

**Comment by nothingismagick** [comment]
@gregory-latinier - can you please check with a new project and new images?

**Comment by gregory-latinier** [comment]
1. You need to commit the yarn.lock since you removed the dependency
2. Is it possible to disable the autoplay?
3. The image I'm uploading and using in the project view page is not displayed the same way in the carousel so it's confusing the user might think the image was cropped or something

image
image
